### PR TITLE
COM-318: Refactor theming of `FilterBar`

### PR DESCRIPTION
**Pull request by juliawegmayr**
Screenshot 2024-01-16 at 14 01 35

### CODE REVIEW

File: packages/admin/admin/src/table/filterbar/FilterBar.tsx [hunk 1]
```diff
@@ -1,46 +1,61 @@
...
-export interface FilterBarProps {
+
+const Root = styled("div", {
+    name: "CometAdminFilterBar",
+    slot: "root",
+    overridesResolver(_, styles) {
+        return [styles.root];
+    },
+})(css`
+    & [class*="CometAdminFormFieldContainer-root"] {
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/FilterBar.tsx:22]: These classes for Comet Admin components no longer exist after the refactor. 
But I'm not sure, how else to do this right now...

**Review by jamesricky — CHANGES_REQUESTED** [review]
Please make sure all deprecation notices stay directly above the deprecated things

**Review by jamesricky — APPROVED** [review]

File: packages/admin/admin/src/table/filterbar/FilterBar.tsx [hunk 2]
```diff
@@ -1,46 +1,64 @@
-import { ComponentsOverrides, Theme } from "@mui/material";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import { ComponentsOverrides, css, styled, Theme, useThemeProps } from "@mui/material/styles";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 
+import { Root as FieldContainerRoot } from "../../form/FieldContainer";
```
Comment (johnnyomair) [inline packages/admin/admin/src/table/filterbar/FilterBar.tsx:5]: @jamesricky isn't it possible to achieve the same thing by using `FieldContainer`? It's strange that the FilterBar needs to know about the elements in FieldContainer...

Comment (jamesricky) [reply packages/admin/admin/src/table/filterbar/FilterBar.tsx:5]: I don't think it's possible, as we need to target a html-element in the selector, not a component with multiple elements... I tried forwarding the `ref` from `FieldContainer` to it's `Root` slot but that didn't change anything 🤷🏼‍♂️

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**